### PR TITLE
Disable dns in multiweave

### PR DESCRIPTION
**Pull request by rade**
Introduce a `--no-dns` option to `weave launch[-router]`. Undocumented, since it's mainly intended for testing. And use that in multiweave, so it works again, post #1065.

### CODE REVIEW

**Comment by tomwilkie** [comment]
If I understand correctly, the only thing stopping multiweave starting is the dns port mapping?  If so, why not disable that (as this change does) for multiweave, but leave all the dns gossip going?   Isn't that what multiweave is suppose to be testing?

**Comment by rade** [comment]
> why not disable that (as this change does) for multiweave, but leave all the dns gossip going?

Didn't realise that was possible. So leave `nameserver.New` and just skip `nameserver.NewDNSServer`?

**Comment by tomwilkie** [comment]
You could do that. Or you could launch both the servers, and just keep the
changes in the weave script to prevent the port mapping.

**Comment by rade** [comment]
> You could do that. Or you could launch both the servers, and just keep the changes in the weave script to prevent the port mapping.

Right. That's a tad obscure if anybody ever does use this option outside testing, since `weave status` would happily report weavedns as running.

**Comment by rade** [comment]
Thinking about it more, I'd rather leave it as is. Disabling DNS should disable all of it. That's the only sensible choice if the option is to be at all useful outside the confines of multiweave.

**Comment by tomwilkie** [comment]
The test failure is #1158

File: prog/weaver/main.go [hunk 1]
```diff
@@ -80,6 +81,7 @@ func main() {
 	mflag.StringVar(&ipsubnetCIDR, []string{"#ipsubnet", "#-ipsubnet", "-ipalloc-default-subnet"}, "", "subnet to allocate within by default, in CIDR notation")
 	mflag.IntVar(&peerCount, []string{"#initpeercount", "#-initpeercount", "-init-peer-count"}, 0, "number of peers in network (for IP address allocation)")
 	mflag.StringVar(&apiPath, []string{"#api", "-api"}, "unix:///var/run/docker.sock", "Path to Docker API socket")
+	mflag.BoolVar(&noDNS, []string{"-no-dns"}, false, "disable DNS server")
```
Comment (tomwilkie) [inline prog/weaver/main.go]: Are you keen on --no-dns?  It doesn't really fit with the rest of the --dns-foo flags.  Could it be --dns-disabled or --dns=false?

Comment (rade) [inline prog/weaver/main.go]: I am quite keen on it. It fits in with the `--no-*` options in the proxy. and `--no-peer-discovery` here.

**Pull request merged**